weave/integrations/openai/openai_sdk.py

```diff
@@ -269,17 +271,54 @@ def _openai_stream_options_is_set(inputs: typing.Dict) -> bool:
     return wrapper
 
 
+# Surprisingly, the async `client.chat.completions.create` does not pass
+# `inspect.iscoroutinefunction`, so we can't dispatch on it and must write
```

Comment:
i wonder if `create` is sort of like our old Op class where it defines a `__acall__` for non-stream and `__aiter__` + `__anext__` for stream? Could we look for that in our op code?

Reply:
I think it's because of their use of these decorators:
https://github.com/openai/openai-python/blob/main/src/openai/resources/completions.py#L571-L572

https://github.com/openai/openai-python/blob/main/src/openai/resources/completions.py#L1030-L1031


If you try to use them, you'll notice that any function decorated with these do not pass `inspect.iscoroutinefunction`

Normally `@overload` would because the pattern is:
```
@overload
async def func(): ...
@overload
async def func(x:int): ...
async def func(x:int,y:int): ...  # this comes last and is not decorated so it will pass

But they don't have that.